scanner/context.go

```diff
@@ -55,7 +55,11 @@ func (c *Context) reset(src []rune) {
 	c.src = src
 	c.tokens = c.tokens[:0]
 	c.resetBuffer()
+	c.isRawFolded = false
 	c.isSingleLine = true
+	c.isLiteral = false
+	c.isFolded = false
+	c.literalOpt = ""
```

Comment:
Individually zeroing fields keeps open the potential for this class of bug in the future if new fields are added to `*Context`. This is why the [original proposal](https://github.com/goccy/go-yaml/issues/316#issuecomment-1312594884) was to use a single assignment to `*c` only retaining the non-zero valued fields.